File: Foody/app/src/main/java/prateek_gupta/foody/util/MyExtensionFunctions.java
package prateek_gupta.foody.util;

import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.Observer;

public class MyExtensionFunctions<T> {
    public void observeOnce(LiveData<T> liveDataObject, LifecycleOwner lifecycleOwner,
                            Observer<T> observer){
        // Calling observe() method to add the Observer object from LifecycleOwner
        // observers list
        liveDataObject.observe(lifecycleOwner, new Observer<T>() {
            @Override
            public void onChanged(T t) {
                // Calling removeObserver() method to remove the Observer object from
                // LifecycleOwner observers list
                liveDataObject.removeObserver(this);

                // Calling onChanged() method to just execute the lambda expression provided
                // for Observer object
                observer.onChanged(t);
            }
        });
    }
}

File: Foody/app/src/main/java/prateek_gupta/foody/util/MyExtensionFunctions.kt
package prateek_gupta.foody.util

import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer

fun <T> LiveData<T>.observeOnce(lifecycleOwner: LifecycleOwner, observer: Observer<T>) {
    // Calling observe() method to add the Observer object from LifecycleOwner
    // observers list
    observe(lifecycleOwner, object : Observer<T> {
        override fun onChanged(t: T?) {
            // Calling removeObserver() method to remove the Observer object from
            // LifecycleOwner observers list
            removeObserver(this)

            // Calling onChanged() method to just execute the lambda expression provided
            // for Observer object
            observer.onChanged(t)
        }
    })
}